examples/list_resources_example.py

```diff
@@ -0,0 +1,8 @@
+from odp_sdk.client import OdpClient
+
+client = OdpClient()
+
+catalog_client = client.catalog
+
+for item in catalog_client.list():
```

Comment:
Maybe add a filter to this example?

Reply:
I will add more in-depth examples in a separate PR after tabular is merged. I like how bare bones this is for now.